src/Drips.sol

```diff
@@ -337,17 +345,24 @@ abstract contract Drips {
         DripsHistory[] memory dripsHistory
     ) internal returns (uint128 amt) {
         uint256 squeezedNum;
-        uint256[] memory squeezedIdxs;
-        (amt, squeezedNum, squeezedIdxs) =
+        uint256[] memory squeezedRevIdxs;
+        bytes32[] memory historyHashes;
+        uint256 currCycleConfigs;
+        (amt, squeezedNum, squeezedRevIdxs, historyHashes, currCycleConfigs) =
```

Comment:
It is kind of interesting with how many different arrays we are dealing in these two functions. The complexity is really high.

1. `dripsHistory` array
   - includes all the information
2.  `squeezedHistoryHashes`
   - only the hashes from the provided dripsHistory
3. `nextSqueezed` array with` currCycleConfigs` as length
   - array can contain elements from outdated cycles :)
4. `squeezedRevIdxs` array with `squeezedNum` as length
   - basically how often we touched `nextSqueezed`
   - hashes in reverse order
5. `squeezedHistoryHashes` array for events length `squeezedNum`
   - array we construct only the present the hashes in the right order


Then it is important to note that we can have three different length
`squeezedNum != historyHashes.length != currCycleConfigs`.

(They can also be the same :) )

Reply:
Yes, and that's terrible. It took me way too long trying to tie all these things together and weird `squeezedRevIdxs` was the best solution I've found :shrug:

Reply:
Yes, it looks good. Just more an overall comment.